This is fixes for the GPU ARM template.

chmitch opened the pull request:
The Nvidia drivers weren't installed on the prior checkin.  This update switches the template VM to the DSVM image which is preconfigured for all the different classes of GPU VMs.

chmitch committed:
add cuda install

chmitch committed:
intermediate testing, on cuda install

chmitch committed:
Switching nvidia driver install to VM Extension

chmitch committed:
fixing bug in vm extension

chmitch committed:
fixing bug in extension

chmitch committed:
yet another attempts

chmitch committed:
switching to a template based on the latest DSVM which has all the nvidia drivers preconfigured

chmitch committed:
fixed parameter reference error

chmitch committed:
remove some cleanup logic seems to be breaking nvidia drivers

chmitch committed:
Reparing some issues in the vm config causing the GPUs to not be addressed.

chmitch committed:
temporary change for local repo testing

chmitch committed:
fixed bug causing incomple creation of config.xml

chmitch committed:
fixed issue with missing trailing line feed in shell script

chmitch committed:
correcting paths to sh script in advance of PR

chmitch committed:
Additional attempt to configure GPU from base Ubuntu server image

chmitch committed:
add -y argument to sudo apt-get install commands

chmitch committed:
adding -y to the ppa driver repository

chmitch committed:
trying to fix bum config file

likamrat commented on deploy/azure/azuredeploy_gpu.json:
```diff
@@ -241,7 +232,7 @@
           "autoUpgradeMinorVersion": true,
           "settings": {
-            "fileUris": ["https://raw.githubusercontent.com/likamrat/foldingathome/master/deploy/azure/install_fah_gpu.sh"],  
+            "fileUris": ["https://raw.githubusercontent.com/chmitch/foldingathome/master/deploy/azure/install_fah_gpu.sh"],  
```
Make sure it's pointing to the right repo

chmitch committed:
why is this config file not getting populated correctly

chmitch committed:
switch gpu slot reference to -1

chmitch committed:
fixing links for pull

chmitch commented:
Updating this PR.   Fixed the errant file path to point to the right repo.   I also fixed the shell script to install nvidia drivers and reverted back to the base ubuntu image.

chmitch committed:
Added parameters files and extended the azure cli documentation to reference the parameter files and also how to override parameters via the cli

chmitch committed:
for local repo dev

likamrat commented on docs/AzureGPU.md:
````diff
@@ -20,10 +20,18 @@ Example: ```az group create -l eastus -n GPU-Folding-COVID19```
 
+# Advanced VM Deployment
+
+The ARM Template leverages a parameters file with preconfigured values for things like username and password.   While you can clone this repo and make the necessary modifications to the files to override the defaults, there is a quicker path leveraging the same ```--parameters``` cli argument.  For example, the following command would do the same deployment with a customized username and password ...
````
Extra space after the first sentence

likamrat commented on docs/AzureGPU.md:
````diff
@@ -20,10 +20,18 @@ Example: ```az group create -l eastus -n GPU-Folding-COVID19```
 
 * Now that you have a new RG created, you can go ahead and deploy the GPU VM in a single command: <br>
 
-```az deployment group create --name "folding" --resource-group "GPU-Folding-COVID19" --template-uri https://raw.githubusercontent.com/likamrat/foldingathome/master/deploy/azure/azuredeploy_gpu.json```
+```az deployment group create --name "folding" --resource-group "GPU-Folding-COVID19" --template-uri https://raw.githubusercontent.com/likamrat/foldingathome/master/deploy/azure/azuredeploy_gpu.json --parameters https://raw.githubusercontent.com/likamrat/foldingathome/master/deploy/azure/azuredeploy_gpu.parameters.json```
 
 **Note: Don't forget to match the resource group name based on the one you created**
 
+# Advanced VM Deployment
+
+The ARM Template leverages a parameters file with preconfigured values for things like username and password.   While you can clone this repo and make the necessary modifications to the files to override the defaults, there is a quicker path leveraging the same ```--parameters``` cli argument.  For example, the following command would do the same deployment with a customized username and password ...
+
+```az deployment group create --name "folding" --resource-group "Folding-COVID19" --template-uri https://raw.githubusercontent.com/likamrat/foldingathome/master/deploy/azure/azuredeploy_gpu.json --parameters adminUsername="<some username>" adminPasswordOrKey="<some strong password>" ```
+
+**Note:  You'll need to replace ```<some username>``` and ```<some strong password>``` with your own values.**
+
 ![](../img/Azure/az_01.png)
 
 ![](../img/Azure/az_02.png)
````
Fix the "Reboot & Shutdown" section at the end of the doc. Currently, it is aligned with Windows.

likamrat reviewed (changes requested):
Fix "Reboot & Shutdown" section

chmitch committed:
deploy to spot instance test

chmitch committed:
changed billing profile so I can change the vm type easily

likamrat reviewed (approved):
LGTM
